src/components/application_manager/src/commands/command_impl.cc

```diff
@@ -193,15 +197,19 @@ bool CommandImpl::ReplaceHMIWithMobileAppId(
         smart_objects::SmartArray* message_array = message.asArray();
         smart_objects::SmartArray::iterator it = message_array->begin();
         for (; it != message_array->end(); ++it) {
-          ReplaceHMIWithMobileAppId(*it);
+          if (!ReplaceHMIWithMobileAppId(*it)) {
+            break;
```

Comment:
shall return false, not true

Reply:
@yang1070 
Done with 866dfe8

Reply:
@yang1070 
Reverted previous changes due to issues with ATF tests. As I found there is a mixed logic for two use cases: first is related to bugfix done for Ford Sync transport switch implementation which requires checking of application with app_id, second is common usage when app_id is replaced during message processing. So I propose to leave code as is for now and create another issue for investigation how this part can be re-factored.